lib/segment/src/vector_storage/raw_scorer.rs

```diff
@@ -915,16 +916,22 @@ where
     }
 
     fn peek_top_all(&self, top: usize) -> Vec<ScoredPointOffset> {
-        let scores = (0..self.point_deleted.len() as PointOffsetType)
+        let point_ids = (0..self.point_deleted.len() as PointOffsetType)
             .take_while(|_| !self.is_stopped.load(Ordering::Relaxed))
             .filter(|point_id| self.check_vector(*point_id))
-            .map(|point_id| {
-                let point_id = point_id as PointOffsetType;
-                ScoredPointOffset {
+            .chunks(64); // batch points to leverage storage sequential access
```

Comment:
Ideally the batch size should be a function of the vector size.

Reply:
Even when batched in chunks of 64 vectors, the "must be consecutive requirement" could be relaxed to have some proportion of it missing and still be able to leverage the sequential reads. E.g. allow some small percentage of the IDs to not be consecutive. WDYT?